batchlib/base.py

```diff
@@ -303,3 +305,29 @@ def __init__(self, input_pattern, input_key, output_key,
                          input_key=input_key, output_key=output_key,
                          input_ndim=input_ndim, output_ndim=output_ndim,
                          target=target)
+
+
+class BatchJobWithSubfolder(BatchJob):
+
+    """ Base class for batch jobs that output into a folder
+    """
+
+    def __init__(self, *args, output_folder="", **kwargs):
+        self.output_folder = output_folder
+
+        super().__init__(*args, **kwargs)
+
+    def to_outputs(self, inputs, folder):
+        names = [os.path.splitext(os.path.split(inp)[1])[0] for inp in inputs]
+        outputs = [os.path.join(folder,
+                                self.output_folder,
+                                name + self.output_ext) for name in names]
```

Comment:
Ok, so you are creating a single sub folder and then all output files go in there.
When we discussed this I thougth you wanted a slightly different pattern where *for each file* a sub-folder is created
and then multiple results could be written to that file, i.e. something like this:
```
ImageXYZ/
  some_result.some_extension
  some_other_result.some_other_extension
```
Anyway, what you are doing here is totally fine, just wanted to point out that I misunderstood it.